cocos/2d/CCScene.cpp

```diff
@@ -211,17 +240,62 @@ void Scene::setPhysics3DDebugCamera(Camera* camera)
 }
 #endif
 
-#if (CC_USE_PHYSICS || (CC_USE_3D_PHYSICS && CC_ENABLE_BULLET_INTEGRATION))
+#if CC_USE_NAVMESH
+void Scene::setNavMeshDebugCamera(Camera *camera)
+{
+    CC_SAFE_RETAIN(camera);
+    CC_SAFE_RELEASE(_navMeshDebugCamera);
+    _navMeshDebugCamera = camera;
+}
+
+void Scene::addChildToNavMesh(Node* child)
+{
+    if (_navMesh)
+    {
+        std::function<void(Node*)> addToNavMeshFunc = nullptr;
+        addToNavMeshFunc = [this, &addToNavMeshFunc](Node* node) -> void
+        {
+            auto agCom = static_cast<NavMeshAgent*>(node->getComponent(NavMeshAgent::getNavMeshAgentComponentName()));
+            if (agCom)
+            {
+                agCom->onEnter();
+            }
+
+            auto obCom = static_cast<NavMeshObstacle*>(node->getComponent(NavMeshObstacle::getNavMeshObstacleComponentName()));
+            if (obCom)
+            {
+                obCom->onEnter();
```

Comment:
`NavMeshAgent ` and `NavMeshObstacle ` are components, the onEnter and onExit is called in Node::OnEnter and Node::OnExit. Why do we call it here?